Source/DataSources/GeometryVisualizer.js

```diff
@@ -408,14 +421,32 @@ define([
                 }
             } else if (updater.isClosed) {
                 if (updater.fillMaterialProperty instanceof ColorMaterialProperty) {
-                    this._closedColorBatches[shadows].add(time, updater);
+                    if (defined(updater.terrainOffsetProperty)) {
+                        this._closedColorBatches[numberOfShadowModes + shadows].add(time, updater);
+                    } else {
+                        this._closedColorBatches[shadows].add(time, updater);
+                    }
                 } else {
-                    this._closedMaterialBatches[shadows].add(time, updater);
+                    // eslint-disable-next-line no-lonely-if
```

Comment:
Why are we disabling the rule here instead of following it? (same comment throughout)

Reply:
I think I was afraid of messing up the if logic here

Reply:
That's what unit tests are for.  You just need to move the if/else up one, please remove these disables.